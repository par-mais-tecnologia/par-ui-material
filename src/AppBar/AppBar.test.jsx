import React from 'react'
import { configure, shallow, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import SeeAppBar from './'
import { MenuItem } from '@material-ui/core'

configure({ adapter: new Adapter() })

const name = 'Rui Cavaleiro'
const menuItems = [
  {
    item: 'SEUS OBJETIVOS',
    onClick: () => {}
  },
  {
    item: 'INTELIGÊNCIA',
    onClick: () => {}
  },
  {
    item: 'INFORMAÇÕES',
    onClick: () => {}
  },
  {
    item: 'CONTATO',
    onClick: () => {}
  }
]

export const testShallow = describe('SeeAppBar', () => {
  const getSeeAppBar = () => (
    <SeeAppBar
      name={name}
      lastUpdateDate={'25/02/2019'}
      menuItems={menuItems}
      handleExit={() => {}}
    />
  )

  const getSeeAppBarWithoutMenuItems = () => (
    <SeeAppBar
      name={name}
      lastUpdateDate={'25/02/2019'}
      menuItems={[]}
      handleExit={() => {}}
    />
  )

  it(`shallow`, () => {
    const wrapper = shallow(getSeeAppBar())
    expect(wrapper).toBeTruthy()
  })

  it('match Snapshot', () => {
    const wrapper = shallow(getSeeAppBar())
    expect(wrapper).toMatchSnapshot()
  })

  it('should have seeMenuMobile', () => {
    const wrapper = mount(getSeeAppBar())
    expect(wrapper.find(`#seeMenuMobile`)).toHaveLength(1)
  })

  it('should not show #seeMenuMobile in Desktop', () => {
    const wrapper = mount(getSeeAppBar())
    expect(wrapper.find(`#seeMenuMobile`).hasClass('dn-l')).toBeTruthy()
  })

  it('should not have <MenuItems> if menuItems prop is empty', () => {
    const wrapper = mount(getSeeAppBarWithoutMenuItems())
    expect(wrapper.find(MenuItem)).toHaveLength(0)
  })

  it('should have <MenuItems> equals menuItems props length', () => {
    const wrapper = mount(getSeeAppBar())
    expect(wrapper.find(MenuItem)).toHaveLength(menuItems.length)
  })
})
