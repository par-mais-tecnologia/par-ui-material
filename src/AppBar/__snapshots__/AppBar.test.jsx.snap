// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SeeAppBar match Snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(SeeAppBar)
    handleExit={[Function]}
    menuItems={
      Array [
        Object {
          "item": "SEUS OBJETIVOS",
          "onClick": [Function],
        },
        Object {
          "item": "INTELIGÊNCIA EM INVESTIMENTOS",
          "onClick": [Function],
        },
        Object {
          "item": "INFORMAÇÕES",
          "onClick": [Function],
        },
        Object {
          "item": "CONTATO",
          "onClick": [Function],
        },
      ]
    }
    mobileFooterBody="Última atualização 
20/09/1989"
    mobileFooterTitle="Ambiente atualizado diariamente"
    name="Rui Cavaleiro"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "classes": Object {
        "exitButton": "SeeAppBar-exitButton-4",
        "iconButton": "SeeAppBar-iconButton-2",
        "menuGutters": "SeeAppBar-menuGutters-3",
        "toolbar": "SeeAppBar-toolbar-1",
      },
      "handleExit": [Function],
      "menuItems": Array [
        Object {
          "item": "SEUS OBJETIVOS",
          "onClick": [Function],
        },
        Object {
          "item": "INTELIGÊNCIA EM INVESTIMENTOS",
          "onClick": [Function],
        },
        Object {
          "item": "INFORMAÇÕES",
          "onClick": [Function],
        },
        Object {
          "item": "CONTATO",
          "onClick": [Function],
        },
      ],
      "mobileFooterBody": "Última atualização 
20/09/1989",
      "mobileFooterTitle": "Ambiente atualizado diariamente",
      "name": "Rui Cavaleiro",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "classes": Object {
          "exitButton": "SeeAppBar-exitButton-4",
          "iconButton": "SeeAppBar-iconButton-2",
          "menuGutters": "SeeAppBar-menuGutters-3",
          "toolbar": "SeeAppBar-toolbar-1",
        },
        "handleExit": [Function],
        "menuItems": Array [
          Object {
            "item": "SEUS OBJETIVOS",
            "onClick": [Function],
          },
          Object {
            "item": "INTELIGÊNCIA EM INVESTIMENTOS",
            "onClick": [Function],
          },
          Object {
            "item": "INFORMAÇÕES",
            "onClick": [Function],
          },
          Object {
            "item": "CONTATO",
            "onClick": [Function],
          },
        ],
        "mobileFooterBody": "Última atualização 
20/09/1989",
        "mobileFooterTitle": "Ambiente atualizado diariamente",
        "name": "Rui Cavaleiro",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
